packages/app-project/src/shared/components/CollectionsModal/components/SelectCollection/SelectCollection.js

```diff
@@ -49,7 +53,12 @@ function SelectCollection ({
     return `${collection.display_name} (${collection.links.owner.display_name})`
   }
 
-  const options = ignorePanoptesFullTextSearch ? collections.filter(collectionNameFilter) : collections
+  /*
+    If the search text is long enough, use fuzzy full-text search. Otherwise, filter collections by display name.
+  */
+  const options = searchText.current.length > MIN_SEARCH_LENGTH
```

Comment:
I don't think this is updating as expected; when I first type 1-3 characters I'd expect it to filter based on display name but the options aren't changing. I think this is because `search.current` is updating in `onTextChange` but it's not triggering a re-render or for whatever reason it's not changing `options` as expected. I'm tinkering and hope to have a solution shortly.

Reply:
You're right. I've pushed a commit that reverts back to `useState`. When I used a ref, the component wouldn't rerender because the returned collections list wasn't changing. The fix is to wait for the search to complete before updating component state, so that the options list updates when the search results are ready.

Reply:
This component is old enough that React hooks (and functional components) didn't exist when I first wrote it. A better implementation now might be to search collections with `react-query` or `swr`.

Reply:
Cool thanks for the quick update! I was messing around similarly, though didn't have it all put together.